atomix/cluster/src/test/java/io/atomix/cluster/messaging/impl/NettyMessagingServiceTest.java

```diff
@@ -152,7 +153,12 @@ public void testSendAsyncToUnresolvable() {
     final String subject = nextSubject();
     final CompletableFuture<Void> response =
         netty1.sendAsync(unresolvable, subject, "hello world".getBytes());
-    assertTrue(response.isCompletedExceptionally());
+
+    assertThrows(
+        CompletionException.class,
+        () -> {
+          netty1.sendAsync(unresolvable, subject, "hello world".getBytes()).join();
+        });
   }
```

Comment:
@oleschoenburg  Since we removed the completeExceptionally when the address is unresolved, I changed the test just to expect for CompletionException, this should be the correct behavior correct? wdyt?

Reply:
Yes, we expect a `CompletionException` now :+1: 
There's a cleaner way to assert this though:

```java
assertThat(response).failsWithin(Duration.ofSeconds(10));
```